File: update-polar-ids.mjs
import { PrismaClient } from '@prisma/client'
import fs from 'fs'
import { parse } from 'csv-parse/sync'

const prisma = new PrismaClient()

async function updateProductsPolarIds() {
  try {
    console.log('📄 Lecture du fichier CSV...')
    
    // Lire le fichier CSV
    const csvContent = fs.readFileSync('/Users/ethanlaloum/Desktop/Freelance/ceramika/Produits_clean.csv', 'utf-8')
    const records = parse(csvContent, {
      columns: true,
      skip_empty_lines: true
    })
    
    console.log(`📊 ${records.length} produits trouvés dans le CSV`)
    
    // Vérifier d'abord combien de produits existent en base
    const existingProducts = await prisma.product.count()
    console.log(`🗄️ ${existingProducts} produits en base de données`)
    
    // Mettre à jour les polarId
    let updatedCount = 0
    let notFoundCount = 0
    
    for (const record of records) {
      try {
        const result = await prisma.product.update({
          where: { id: record.id },
          data: { polarId: record.polarId }
        })
        updatedCount++
        
        if (updatedCount % 50 === 0) {
          console.log(`⏳ Mis à jour: ${updatedCount}/${records.length}`)
        }
      } catch (error) {
        if (error.code === 'P2025') {
          // Produit non trouvé
          notFoundCount++
        } else {
          console.error(`❌ Erreur pour le produit ${record.id}:`, error.message)
        }
      }
    }
    
    console.log(`✅ Mise à jour terminée:`)
    console.log(`   - ${updatedCount} produits mis à jour`)
    console.log(`   - ${notFoundCount} produits non trouvés en base`)
    
    // Vérification finale
    const productsWithPolarId = await prisma.product.count({
      where: {
        polarId: {
          not: null,
          not: ''
        }
      }
    })
    
    console.log(`🎯 ${productsWithPolarId} produits ont maintenant un polarId`)
    
    // Exemples de produits mis à jour
    const sampleProducts = await prisma.product.findMany({
      where: {
        polarId: {
          not: null,
          not: ''
        }
      },
      select: {
        id: true,
        name: true,
        polarId: true
      },
      take: 5
    })
    
    console.log('📋 Exemples de produits avec polarId:')
    sampleProducts.forEach(p => {
      console.log(`   - ${p.name}: ${p.polarId}`)
    })
    
  } catch (error) {
    console.error('❌ Erreur lors de la mise à jour:', error)
  } finally {
    await prisma.$disconnect()
  }
}

updateProductsPolarIds()

File: update-polar-ids.py
#!/usr/bin/env python3
import csv
import asyncio
import asyncpg
import os
from dotenv import load_dotenv

# Charger les variables d'environnement
load_dotenv()

async def update_products_polar_ids():
    """Met à jour les polarId des produits en base depuis le CSV"""
    
    # Connexion à la base de données
    database_url = os.getenv('DATABASE_URL')
    if not database_url:
        print("❌ DATABASE_URL non trouvée dans les variables d'environnement")
        return
    
    try:
        conn = await asyncpg.connect(database_url)
        print("✅ Connexion à la base de données établie")
        
        # Lire le fichier CSV
        csv_file_path = "/Users/ethanlaloum/Desktop/Freelance/ceramika/Produits_clean.csv"
        products_data = []
        
        with open(csv_file_path, 'r', encoding='utf-8') as file:
            reader = csv.DictReader(file)
            for row in reader:
                products_data.append({
                    'id': row['id'],
                    'polarId': row['polarId']
                })
        
        print(f"📄 Lecture de {len(products_data)} produits depuis le CSV")
        
        # Mettre à jour chaque produit
        updated_count = 0
        for product in products_data:
            try:
                result = await conn.execute(
                    "UPDATE products SET \"polarId\" = $1 WHERE id = $2",
                    product['polarId'], product['id']
                )
                if result.endswith('1'):  # 1 ligne affectée
                    updated_count += 1
            except Exception as e:
                print(f"❌ Erreur pour le produit {product['id']}: {e}")
        
        print(f"✅ {updated_count} produits mis à jour avec leurs polarId")
        
        # Vérifier le résultat
        products_with_polar_id = await conn.fetchval(
            "SELECT COUNT(*) FROM products WHERE \"polarId\" IS NOT NULL AND \"polarId\" != ''"
        )
        
        print(f"📊 Produits avec polarId en base: {products_with_polar_id}")
        
        await conn.close()
        
    except Exception as e:
        print(f"❌ Erreur lors de la mise à jour: {e}")

if __name__ == "__main__":
    asyncio.run(update_products_polar_ids())
